src/lib.rs

```diff
@@ -108,27 +106,48 @@ pub enum FoundCrate {
 /// The returned crate name is sanitized in such a way that it is a valid rust identifier. Thus,
 /// it is ready to be used in `extern crate` as identifier.
 pub fn crate_name(orig_name: &str) -> Result<FoundCrate, Error> {
-    let manifest_dir =
-        PathBuf::from(env::var("CARGO_MANIFEST_DIR").map_err(|_| Error::CargoManifestDirNotSet)?);
+    struct Cache {
+        manifest_dir: String,
+        manifest_path: PathBuf,
+        manifest: Table,
+    }
 
-    let cargo_toml_path = manifest_dir.join("Cargo.toml");
+    static CACHE: OnceCell<Cache> = OnceCell::new();
+    let cache = CACHE.get_or_try_init(|| {
+        let manifest_dir =
+            env::var("CARGO_MANIFEST_DIR").map_err(|_| Error::CargoManifestDirNotSet)?;
 
-    if !cargo_toml_path.exists() {
-        return Err(Error::NotFound(manifest_dir.into()));
-    }
+        let manifest_path = Path::new(&manifest_dir).join("Cargo.toml");
+
+        if !manifest_path.exists() {
+            return Err(Error::NotFound(manifest_dir.into()));
+        }
+
+        let manifest = open_cargo_toml(&manifest_path)?;
+
+        Ok(Cache {
+            manifest_dir,
+            manifest_path,
+            manifest,
+        })
+    })?;
 
-    let cargo_toml = open_cargo_toml(&cargo_toml_path)?;
+    let manifest_dir = env::var("CARGO_MANIFEST_DIR").map_err(|_| Error::CargoManifestDirNotSet)?;
+    assert_eq!(
+        manifest_dir, cache.manifest_dir,
+        "CARGO_MANIFEST_DIR must not change within one compiler process"
+    );
 
-    extract_crate_name(orig_name, cargo_toml, &cargo_toml_path)
+    extract_crate_name(orig_name, &cache.manifest, &cache.manifest_path)
 }
 
 /// Make sure that the given crate name is a valid rust identifier.
 fn sanitize_crate_name<S: AsRef<str>>(name: S) -> String {
-    name.as_ref().replace("-", "_")
+    name.as_ref().replace('-', "_")
```

Comment:
```suggestion
    name.as_ref().replace("-", "_")
```

Reply:
Clippy will complain about that one. Is there a reason you prefer it?

Reply:
Hmm :P Okay fine. Just "looks" weird.